src/components/productoverview/ProductSorting.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+    <select data-test="sorting-select"
+            name="sort"
+            id="sort"
+            v-model="sortBy">
+        <option value="">--</option>
+        <option value="newest">Newest</option>
+        <option value="oldest">Oldest</option>
+    </select>
+</template>
+
+<script>
+export default {
+  data: () => ({
+    sortBy: '',
+  }),
+  created() {
+    this.sortBy = this.$route.query.sort || '';
+  },
+
+  watch: {
+    sortBy(sortBy) {
+      if (sortBy.length > 0) {
+        this.$router.replace({ query: { sort: sortBy } });
+      } else {
+        this.$router.replace('');
+      }
```

Comment:
Does this work when there are other parameters in the URL? In the product overview page there will be pagination, filters, sorting, products per page selectors and search. All of them will be using the query parameters, so we could have a URL with ?sort=newest&color=red&size=M&p=3&ppp=24&search=wool%20jacket

Could you please check if it works under these conditions?